### PR TITLE
Add demo environment infrastructure

### PR DESCRIPTION
**Pull request by giorgiosironi**
First step of https://github.com/libero/libero/issues/166

### CODE REVIEW

File: tf/blog-articles.tf [hunk 1]
```diff
@@ -47,8 +47,8 @@ output "credentials_blog_articles_secret" {
 }
 
 resource "aws_iam_policy" "blog_articles_s3_write" {
-  name        = "blogArticlesS3Write"
-  path        = "/"
+  name        = "${var.env}BlogArticlesS3Write"
```
Comment (thewilkybarkid) [inline tf/blog-articles.tf]: Demo's only for JATS content (for now), so won't need multiple content stores.

Comment (giorgiosironi) [reply tf/blog-articles.tf]: The Terraform model seems too monolithic to turn things on and off :thinking: but also, the smoke tests are performed on `unstable` so there's the usual argument for keeping it identical. Could remove it from there as well...

File: .travis.yml [hunk 2]
```diff
@@ -16,13 +16,18 @@ before_install:
...
+  - rm -rf .terraform && ./terraform init --backend-config="key=demo/terraform.tfstate" -input=false
+  - ./terraform plan -var end=demo -out /tmp/demo.plan
 
 deploy:
 - provider: script
   script:
-    - ./terraform apply /tmp/terraform.plan
+    - rm -rf .terraform && ./terraform init --backend-config="key=unstable/terraform.tfstate" -input=false
+    - ./terraform apply /tmp/unstable.plan
+    - rm -rf .terraform && ./terraform init --backend-config="key=demo/terraform.tfstate" -input=false
```
Comment (GiancarloFusiello) [inline .travis.yml]: Can you move this to a tag build?

File: tf/providers.tf [hunk 3]
```diff
@@ -8,7 +8,7 @@ provider "local" {
 terraform {
   backend "s3" {
     bucket = "libero-terraform"
-    # specify with terraform init --backend-config="key=unstable/terraform.tfstate" to make it dynamic
+    # specify with terraform init --backend-config="key=.../terraform.tfstate" to make it dynamic
```
Comment (GiancarloFusiello) [inline tf/providers.tf]:
```suggestion
    # specify with terraform init --backend-config="key=<env>/terraform.tfstate" to make it dynamic
```

**Pull request merged**